models/fates/R/write.configs.FATES.R

```diff
@@ -52,56 +45,60 @@ write.config.FATES <- function(defaults, trait.values, settings, run.id){
 
    ## SITE INFO --> DOMAIN FILE (lat/lon)
    #  - should also store this in the refcase directory for PEcAn so we can grab from there, and not the PEcAn package
-   gridres <- 0.125  ## ultimately this should be a variable
-   lat <- as.numeric(site$lat)
-   lon <- (as.numeric(site$lon) + 360) %% 360 ## make sure coords in 0-360 range, not negative
-   domain.default <- system.file("domain.lnd.1x1pt-brazil_navy.090715.nc",package="PEcAn.FATES")
-   domain.file <- file.path(local.rundir,paste0("domain.lnd.",site_name,".nc"))
-   file.copy(domain.default,domain.file)
-   domain.nc <- ncdf4::nc_open(domain.file,write=TRUE)
-   ncdf4::ncvar_put(nc=domain.nc, varid='xc', vals=lon)
-   ncdf4::ncvar_put(nc=domain.nc, varid='yc', vals=lat)
-   ncdf4::ncvar_put(nc=domain.nc, varid='xv', vals=lon+c(-1,1,1,-1)*gridres)
-   ncdf4::ncvar_put(nc=domain.nc, varid='yv', vals=lat+c(-1,-1,1,1)*gridres)
-   ncdf4::ncvar_put(nc=domain.nc, varid='area', vals=(2*gridres*pi/180)^2)   
-   ncdf4::nc_close(domain.nc)
+   #gridres <- 0.125  ## ultimately this should be a variable
+   #lat <- as.numeric(site$lat)
+   #lon <- (as.numeric(site$lon) + 360) %% 360 ## make sure coords in 0-360 range, not negative
+   #domain.default <- system.file("domain.lnd.360x720_gswp3.0v1.c170606.nc",package="PEcAn.FATES")
+   #domain.file <- file.path(local.rundir,paste0("domain.lnd.fv0.9x1.25_gx1v7_",site_name,".nc"))
+   #file.copy(domain.default,domain.file)
+   #domain.nc <- ncdf4::nc_open(domain.file,write=TRUE)
+   #ncdf4::ncvar_put(nc=domain.nc, varid='xc', vals=lon)
+   #ncdf4::ncvar_put(nc=domain.nc, varid='yc', vals=lat)
+   #ncdf4::ncvar_put(nc=domain.nc, varid='xv', vals=lon+c(-1,1,1,-1)*gridres)
+   #ncdf4::ncvar_put(nc=domain.nc, varid='yv', vals=lat+c(-1,-1,1,1)*gridres)
+   #ncdf4::ncvar_put(nc=domain.nc, varid='area', vals=(2*gridres*pi/180)^2)   
+   #ncdf4::nc_close(domain.nc)
    
    ## SURF - should also store this in the refcase directory for PEcAn so we can grab from there, and not the PEcAn package
-   surf.default <- system.file("surfdata_1x1_brazil_16pfts_Irrig_CMIP6_simyr2000_c171214.nc",package = "PEcAn.FATES")
-   surf.file    <- file.path(local.rundir,paste0("surfdata_",site_name,"_simyr2000.nc"))
-   file.copy(surf.default,surf.file)
-   Sys.chmod(surf.file)
-   surf.nc <- ncdf4::nc_open(surf.file,write=TRUE)
-   ncdf4::ncvar_put(nc=surf.nc, varid='LONGXY', vals=lon)
-   ncdf4::ncvar_put(nc=surf.nc, varid='LATIXY', vals=lat)
-   ncdf4::nc_close(surf.nc)   
+   #surf.default <- system.file("surfdata_0.9x1.25_hist_16pfts_Irrig_CMIP6_simyr2000_c190214.nc",package = "PEcAn.FATES")
```

Comment:
Same for SURFDATA file? Is this no longer needed?

Reply:
SURFDATA is still needed. At the moment, this is set directly in template.job. We are not sure if "SURFDATA file" is assumed to be given directly as a setting variable (from Bety) in PEcAn?  Or PEcAn want "SURFDATA file" to be automatically generated using a default global file and then modify the variables (e.g., soil texture) using the exiting tools from CTSM (https://github.com/ESCOMP/CTSM/tree/master/tools/site_and_regional). I think it would be up to you to decide which way would be best. We could then implement the changes accordingly.

Reply:
So either way there definitely shouldn't be any requirement to manually run CTSM tools before or after a PEcAn write.config.  I think in the old set up I was just setting the lat/lon of the run. I never got around to coupling soil texture, but that can be provided by pecan here: https://github.com/PecanProject/pecan/blob/e43e8ff6ef31cd0a908e60c3b51a342dd4f01ecd/modules/data.land/R/soil2netcdf.R#L34

Like met, PEcAn supports multiple soil texture input sources, and can even ensemble-sample the uncertainties in some of these (e.g., SoilGrids)

Reply:
The ability to change soil texture and make ensemble-sample & experiments of it sounds really cool. Then, could we assume that PEcAn will anyway provide a default surface data file (in netcdf), and also some "observed" soil or vegetation properties (like soil texture, monthly LAI etc...), and then the "write.config." will manage to modify the default surface data file with "observed" or "perturbed" values? Modify an existing netcdf file with new values should be easy and straightforward. We can do this together with modifying the fates parameter file.